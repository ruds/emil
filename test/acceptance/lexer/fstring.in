f""
F""
f""""""
F""""""
fr"foo()foo"
fR"foo()foo"
Fr"foo()foo"
FR"foo()foo"
f"\$foo"
f"$foo"
f"\${foo}"
f"${foo}"
f"""\$foo"""
f"""$foo"""
f"""\${foo}"""
f"""${foo}"""
fr"(\$foo)"
fr"($foo)"
fr"(\${foo})"
fr"(${foo})"
"$foo"
"""$foo"""
R"($foo)"
f"${"foo"}"
f"""${"""foo"""}"""
fr"foo(${r"foo(foo)foo"})foo"
f"foo + 1 =${let
  val a = foo
  val b = 1
in
  a + b (* comments are allowed *)
end} when foo =$foo."
f"""foo + 1 =${let
  val a = foo
  val b = 1
in
  a + b (* comments are allowed *)
end} when foo =$foo."""
fr"(foo + 1 =${let
  val a = foo
  val b = 1
in
  a + b (* comments are allowed *)
end} when foo =$foo.)"
f"foo${ { foo, bar } } bar"
f"foo =
(*xx*)
f"foo =$foo.
(*xx*)
f"""this is the string that${ " never ends" """ """
