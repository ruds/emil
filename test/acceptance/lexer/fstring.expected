0001 FSTRING f""
0002 FSTRING F""
0003 FSTRING f""""""
0004 FSTRING F""""""
0005 FSTRING fr"foo()foo"
0006 FSTRING fR"foo()foo"
0007 FSTRING Fr"foo()foo"
0008 FSTRING FR"foo()foo"
0009 FSTRING 24666F6F f"\$foo"
0010 FSTRING f"
0010 FSTRING_IVAR 666F6F $foo
0010 FSTRING_CONT "
0011 FSTRING 247B666F6F7D f"\${foo}"
0012 FSTRING f"
0012 FSTRING_IEXPR_S ${
0012 ID_WORD 666F6F foo
0012 FSTRING_IEXPR_F }
0012 FSTRING_CONT "
0013 FSTRING 24666F6F f"""\$foo"""
0014 FSTRING f"""
0014 FSTRING_IVAR 666F6F $foo
0014 FSTRING_CONT """
0015 FSTRING 247B666F6F7D f"""\${foo}"""
0016 FSTRING f"""
0016 FSTRING_IEXPR_S ${
0016 ID_WORD 666F6F foo
0016 FSTRING_IEXPR_F }
0016 FSTRING_CONT """
0017 FSTRING 5C fr"(\
0017 FSTRING_IVAR 666F6F $foo
0017 FSTRING_CONT )"
0018 FSTRING fr"(
0018 FSTRING_IVAR 666F6F $foo
0018 FSTRING_CONT )"
0019 FSTRING 5C fr"(\
0019 FSTRING_IEXPR_S ${
0019 ID_WORD 666F6F foo
0019 FSTRING_IEXPR_F }
0019 FSTRING_CONT )"
0020 FSTRING fr"(
0020 FSTRING_IEXPR_S ${
0020 ID_WORD 666F6F foo
0020 FSTRING_IEXPR_F }
0020 FSTRING_CONT )"
0021 STRING 24666F6F "$foo"
0022 STRING 24666F6F """$foo"""
0023 STRING 24666F6F R"($foo)"
0024 FSTRING f"
0024 FSTRING_IEXPR_S ${
0024 STRING 666F6F "foo"
0024 FSTRING_IEXPR_F }
0024 FSTRING_CONT "
0025 FSTRING f"""
0025 FSTRING_IEXPR_S ${
0025 STRING 666F6F """foo"""
0025 FSTRING_IEXPR_F }
0025 FSTRING_CONT """
0026 FSTRING fr"foo(
0026 FSTRING_IEXPR_S ${
0026 STRING 666F6F r"foo(foo)foo"
0026 FSTRING_IEXPR_F }
0026 FSTRING_CONT )foo"
0027 FSTRING 666F6F202B2031203D f"foo + 1 =
0027 FSTRING_IEXPR_S ${
0027 KW_LET let
0028 KW_VAL val
0028 ID_WORD 61 a
0028 ID_OP 3D =
0028 ID_WORD 666F6F foo
0029 KW_VAL val
0029 ID_WORD 62 b
0029 ID_OP 3D =
0029 ILITERAL 1 1
0030 KW_IN in
0031 ID_WORD 61 a
0031 ID_OP 2B +
0031 ID_WORD 62 b
0032 KW_END end
0032 FSTRING_IEXPR_F }
0032 FSTRING_CONT 207768656E20666F6F203D  when foo =
0032 FSTRING_IVAR 666F6F $foo
0032 FSTRING_CONT 2E ."
0033 FSTRING 666F6F202B2031203D f"""foo + 1 =
0033 FSTRING_IEXPR_S ${
0033 KW_LET let
0034 KW_VAL val
0034 ID_WORD 61 a
0034 ID_OP 3D =
0034 ID_WORD 666F6F foo
0035 KW_VAL val
0035 ID_WORD 62 b
0035 ID_OP 3D =
0035 ILITERAL 1 1
0036 KW_IN in
0037 ID_WORD 61 a
0037 ID_OP 2B +
0037 ID_WORD 62 b
0038 KW_END end
0038 FSTRING_IEXPR_F }
0038 FSTRING_CONT 207768656E20666F6F203D  when foo =
0038 FSTRING_IVAR 666F6F $foo
0038 FSTRING_CONT 2E ."""
0039 FSTRING 666F6F202B2031203D fr"(foo + 1 =
0039 FSTRING_IEXPR_S ${
0039 KW_LET let
0040 KW_VAL val
0040 ID_WORD 61 a
0040 ID_OP 3D =
0040 ID_WORD 666F6F foo
0041 KW_VAL val
0041 ID_WORD 62 b
0041 ID_OP 3D =
0041 ILITERAL 1 1
0042 KW_IN in
0043 ID_WORD 61 a
0043 ID_OP 2B +
0043 ID_WORD 62 b
0044 KW_END end
0044 FSTRING_IEXPR_F }
0044 FSTRING_CONT 207768656E20666F6F203D  when foo =
0044 FSTRING_IVAR 666F6F $foo
0044 FSTRING_CONT 2E .)"
0045 FSTRING 666F6F f"foo
0045 FSTRING_IEXPR_S ${
0045 LBRACE {
0045 ID_WORD 666F6F foo
0045 COMMA ,
0045 ID_WORD 626172 bar
0045 RBRACE }
0045 FSTRING_IEXPR_F }
0045 FSTRING_CONT 20626172  bar"
0046 ERROR
0048 FSTRING 666F6F203D f"foo =
0048 FSTRING_IVAR 666F6F $foo
0048 ERROR
0050 FSTRING 746869732069732074686520737472696E672074686174 f"""this is the string that
0050 FSTRING_IEXPR_S ${
0050 STRING 206E6576657220656E6473 " never ends"
0050 STRING 20 """ """
0051 ERROR
0051 END_OF_FILE
